mllib/src/test/scala/org/apache/spark/mllib/stat/HypothesisTestSuite.scala

```diff
@@ -254,4 +254,115 @@ class HypothesisTestSuite extends SparkFunSuite with MLlibTestSparkContext {
     assert(rCompResult.statistic ~== rKSStat relTol 1e-4)
     assert(rCompResult.pValue ~== rKSPVal relTol 1e-4)
   }
+
```

Comment:
The edge case of partitions with no elements does indeed exist, which is why we have that odd check for whether a result after folding in a partition matches the initial accumulator. The test involving `nonOverlap1P` and `nonOverlap2P` should test for situations where a partition has only 1 sample. I'll break these out into separate tests, since they probably deserve to be singled out.